### ISSUE TITLE
Application field throws error (can't submit form) when required is false

### ISSUE THREAD

**Issue by tripflex** [Automattic/WP-Job-Manager#2324]
**Describe the bug**
If a user decides to set the `application` field as optional (setting required to `false`) the form will not allow them to submit at all, due to the specific validation handling found below, that does not check the `required` flag in any way:
https://github.com/Automattic/WP-Job-Manager/blob/trunk/includes/forms/class-wp-job-manager-form-submit-job.php#L516-L547

Users **can** just use the `job_manager_application_field_skip_email_url_validation` filter to return true, but this would then remove all validation regardless of whether the field is required or not.

**To Reproduce**
1. Set `required` to false for `application` field through filter or Field Editor
2. Goto submit page, leave Application field blank
3. Form can not be submitted and "Please enter a valid application email address or URL" error is shown

Basically we need to add a check in the `validate_fields` method to see if `application` field is required or not
https://github.com/Automattic/WP-Job-Manager/blob/0a2d4e337663067106331cb86fe5a033d02220cc/includes/forms/class-wp-job-manager-form-submit-job.php#L516
